apps/dcos_dns/src/dcos_dns_router.erl

```diff
@@ -15,12 +15,33 @@
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = dcos_dns_app:parse_upstream_name(Name),
     find_upstream(Labels);
-upstreams_from_questions([Question|Others]) ->
-    %% There is more than one question. This is beyond our capabilities at the moment
-    dcos_dns_metrics:update([dcos_dns, ignored_questions], length(Others), ?COUNTER),
-    Result = upstreams_from_questions([Question]),
-    lager:debug("~p will be forwarded to ~p", [Others, Result]),
-    Result.
+upstreams_from_questions(Questions) ->
+    AllUpstreams = [upstreams_from_questions([Q]) || Q <- Questions],
+    case lists:usort(AllUpstreams) of
+        [Upstream] ->
+            Upstream;
+        MixedUpstreams ->
```

Comment:
Let's make the message more readable. I think it's fine to print erlang terms at the end of the message (in this case erlang term can be truncated, but human readable message will be printed). Here are my suggestion:

```erlang
        _Upstreams ->
            lager:warning(
                "DNS queries with mixed-upstream questions are not supported, "
                "the query will be resolved through internal DNS server: ~p",
                [Questions]),
            lists:foreach(fun (Zone) ->
                prometheus_counter:inc(
                    dns, forwarder_failures_total,
                    [Zone], 1)
            end, [Z || {U, Z} <- AllUpstreams, U =/= internal]),
            {internal, <<".">>}
```